app/src/org/commcare/tasks/SaveToDiskTask.java

```diff
@@ -126,7 +127,7 @@ protected ResultAndError<SaveStatus> doTaskBackground(Void... nothing) {
             // Likely a user level issue, so send error to HQ as a app build error
             XPathErrorLogger.INSTANCE.logErrorToCurrentApp(cleanedMessage);
 
-            return new ResultAndError<>(SaveStatus.SAVE_ERROR, cleanedMessage);
+            return new ResultAndError<>(SaveStatus.SAVE_UNRECOVERABLE_ERROR, cleanedMessage);
```

Comment:
Also would you be able to provide more details around the error pathway you think the [crash linked in the PR](https://github.com/dimagi/commcare-android/pull/2803) will surface as in this method.